design-proposals/instancetype.kubevirt.io/extend-preferSpread.md

```diff
@@ -0,0 +1,325 @@
+# Overview
+
+The `preferSpread` `preferredCPUTopology` was introduced with KubeVirt v1.1 and allows for vCPUs provided by an instance type to be spread across guest visible sockets and cores using a configurable ratio that defaults to 2.
+
+For example:
+
+```yaml
+
+$ ./cluster-up/kubectl.sh apply -f -<<EOF
+---
+apiVersion: instancetype.kubevirt.io/v1beta1
+kind: VirtualMachineInstancetype
+metadata:
+  name: spread
+spec:
+  cpu:
+    guest: 4
+  memory:
+    guest: 1Gi
+---
+apiVersion: instancetype.kubevirt.io/v1beta1
+kind: VirtualMachinePreference
+metadata:
+  name: spread
+spec:
+  cpu:
+    preferredCPUTopology: preferSpread
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm-spread
+spec:
+  instancetype:
+    name: spread
+    kind: VirtualMachineInstancetype
+  preference:
+    name: spread
+    kind: VirtualMachinePreference
+  runStrategy: Always
+  template:
+    spec:
+      domain:
+        devices: {}
+      volumes:
+        - containerDisk:
+            image: quay.io/containerdisks/fedora:39
+          name: containerdisk
+EOF
+[..]
+$ ./cluster-up/kubectl.sh get vmi/vm-spread -o json | jq .spec.domain.cpu
+{
+  "cores": 2,
+  "model": "host-model",
+  "sockets": 2,
+  "threads": 1
+}
+```
+
+This design proposal aims to extend this implementation by introducing new configurables within a preference to control how vCPUs are spread with the goal of allowing vCPUs to also be spread to threads.
+
+## Motivation
+
+The current implementation of the `preferSpread` `preferredCPUTopology` is limited to spreading vCPUs across sockets and cores of the guest CPU topology. While a useful starting point it would be beneficial to also allow these vCPUs to be spread across all parts of the guest visible CPU topology including threads in order to mimic SMT within the guest.
```

Comment:
FWIW why is there the redundancy of the `prefer` string between the key and value?

Reply:
See later in the proposal, we should clean this up and drop the `prefer` so this becomes just `Spread`.